lib/oli_web/controllers/payment_controller.ex

```diff
@@ -90,18 +90,9 @@ defmodule OliWeb.PaymentController do
     # after already paying.  This will simply redirect them to their course.
   end
 
-  defp determine_product(section) do
-    if is_nil(section.blueprint_id) do
-      section
-    else
-      section.blueprint
-    end
-  end
-
-  defp determine_cost(section) do
-    section = Oli.Repo.preload(section, [:institution, :blueprint])
-    product = determine_product(section)
-    Oli.Delivery.Paywall.calculate_product_cost(product, section.institution)
+  defp determine_enrollable_cost(section) do
```

Comment:
I get your point, but I don't know of the top of my head how that would behave for products that have a discount applied and the sections that have parted from the product have not had their amount updated to match. Will investigate and reach out to see that we're not breaking other functionality if we use the amount directly. I need to inform myself on the intention of discounts for sections

Reply:
Discussed offline. We're OK with always taking the value directly from sections. When a section is created from a product, the discounts are applied so that the new section cost = product cost - discounts. No need to re-calculate it.